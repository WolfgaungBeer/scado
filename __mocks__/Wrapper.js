import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Theme, defaultTheme } from '../scado';

const mockStore = configureStore();
const store = mockStore({
    scado: {
        theme: defaultTheme,
        toasts: [
            { id: '1', type: 'success', content: 'Text', onClose: () => {} },
            { id: '2', type: 'error', content: 'Text', onClose: () => {} },
        ],
    },
});


const Wrapper = ({ children }) => ( // eslint-disable-line
    <Provider store={store}>
        <Theme>{children}</Theme>
    </Provider>
);

export default Wrapper;
