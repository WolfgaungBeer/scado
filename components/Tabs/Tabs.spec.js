import React from 'react';
import renderer from 'react-test-renderer';
import Wrapper from '../../__mocks__/Wrapper';
import { Tabs, Tab } from './';

it('renders the Text Component correctly', () => {
    const tree = renderer.create(
        <Wrapper>
            <Tabs>
                <Tab active onClick={() => {}}>tab 1</Tab>
                <Tab onClick={() => {}}>tab 2</Tab>
                <Tab onClick={() => {}}>tab 3</Tab>
            </Tabs>
        </Wrapper>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
