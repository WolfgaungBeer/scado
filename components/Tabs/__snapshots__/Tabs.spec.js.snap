// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Text Component correctly 1`] = `
<div
  className="sc-bwzfXH csttzL"
>
  <div
    className="sc-htpNat hARGnU"
    onClick={[Function]}
  >
    tab 1
  </div>
  <div
    className="sc-htpNat bOfeLP"
    onClick={[Function]}
  >
    tab 2
  </div>
  <div
    className="sc-htpNat bOfeLP"
    onClick={[Function]}
  >
    tab 3
  </div>
</div>
`;
